sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala

```diff
@@ -448,11 +448,18 @@ abstract class AbstractGpuCoalesceIterator(
                     s" are in this partition. Please try increasing your partition count.")
               case RequireSingleBatchWithFilter(filterExpression) =>
                 // filter what we had already stored
-                val filteredDown = GpuFilter(concatAllAndPutOnGPU(), filterExpression)
+                val filteredDown = withResource(concatAllAndPutOnGPU()) { concatCB =>
+                  GpuFilter(concatCB, filterExpression)
+                }
                 closeOnExcept(filteredDown) { _ =>
-                  // filter the incoming batch as well
-                  closeOnExcept(GpuFilter(cb, filterExpression)) { filteredCb =>
-                    cb = null // null out `cb` to prevent multiple close calls
+                  // filter the incoming batch if not be filtered.
+                  val filteredCb = if (inputFilterExpression.isEmpty) {
```

Comment:
Why filter only if the filter isEmpty?

Reply:
When the filter expression is defined, the batch will be filtered at line 424.
```
      var cb = if (inputFilterExpression.isDefined) {
        // If we have reached the cuDF limit once, proactively filter batches
        // after that first limit is reached.
        withResource(cbFromIter)(GpuFilter(_, inputFilterExpression.get))
      } else {
        cbFromIter
      }
```